Source/Core/Absy.cs

```diff
@@ -3669,6 +3675,11 @@ public String ErrorMessage
 
     public QKeyValue Attributes { get; set; }
 
+    public bool CanAlwaysAssume ()
+    {
+      return QKeyValue.FindBoolAttribute(this.Attributes, "always_assume");
```

Comment:
You can also check that the requires/ensures is Free within this method so that the caller does not have to check for Free.